bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,252 @@
+                                sslContextFactory.setSslContext(sslContext);
+                            } catch (Exception ex) {
+                                throw new RuntimeException(
+                                        "Cannot create an TLS context for the endpoint '" + endpoint + "'!", ex);
+                            }
+                        }
+                    }
+                    String excludeCipherSuites[] = { "^.*_(MD5)$" };
+                    sslContextFactory.setExcludeCipherSuites(excludeCipherSuites);
+
+                    HttpClient httpClient = new HttpClient(sslContextFactory);
+                    final QueuedThreadPool queuedThreadPool = createThreadPool(consumerName, minThreadsCustom,
+                            maxThreadsCustom, keepAliveTimeoutCustom);
+
+                    httpClient.setExecutor(queuedThreadPool);
+
+                    try {
+                        httpClient.start();
+                    } catch (Exception e) {
+                        logger.error("Could not start jetty client", e);
+                        throw new RuntimeException("Could not start jetty client", e);
+                    }
+
+                    return httpClient;
+                }
+            });
+        } catch (PrivilegedActionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof RuntimeException) {
+                throw (RuntimeException) cause;
+            } else {
+                throw new RuntimeException(cause);
+            }
+        }
+    }
+
+    private QueuedThreadPool createThreadPool(String consumerName, int minThreads, int maxThreads,
+            int keepAliveTimeout) {
+        QueuedThreadPool queuedThreadPool = new QueuedThreadPool(maxThreads, minThreads, keepAliveTimeout);
+        queuedThreadPool.setName(consumerName);
+        queuedThreadPool.setDaemon(true);
+        return queuedThreadPool;
+    }
+
+    private void checkConsumerName(String consumerName) {
+        Objects.requireNonNull(consumerName, "consumerName must not be null");
+        if (consumerName.length() < MIN_CONSUMER_NAME_LENGTH) {
+            throw new IllegalArgumentException(
+                    "consumerName " + consumerName + " too short, minimum " + MIN_CONSUMER_NAME_LENGTH);
+        }
+        if (consumerName.length() > MAX_CONSUMER_NAME_LENGTH) {
+            throw new IllegalArgumentException(
+                    "consumerName " + consumerName + " too long, maximum " + MAX_CONSUMER_NAME_LENGTH);
+        }
+        if (!CONSUMER_NAME_PATTERN.matcher(consumerName).matches()) {
+            throw new IllegalArgumentException(
+                    "consumerName " + consumerName + " contains illegal character, allowed only [a-zA-Z0-9_-]");
+        }
+    }
+
+    @Reference(service = TrustManagerProvider.class, cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)
```

Comment:
`service` attribute is not necessary. If it is not specified, the type of the service to bind is based upon the type of the first argument of the method being annotated.

Reply:
I like the explicit approach